synapse/tools/storm.py

```diff
@@ -0,0 +1,190 @@
+import sys
+import asyncio
+import logging
+import argparse
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+import synapse.telepath as s_telepath
+
+import synapse.lib.cli as s_cli
+import synapse.lib.node as s_node
+import synapse.lib.time as s_time
+import synapse.lib.output as s_output
+
+logger = logging.getLogger(__name__)
+
+ERROR_COLOR = '#ff0066'
+WARNING_COLOR = '#f4e842'
+NODEEDIT_COLOR = "lightblue"
+
+welcome = '''
+Welcome to the Storm interpreter!
+
+Local interpreter (non-storm) commands may be executed with a ! prefix:
+    Use !quit to exit.
+    Use !help to see local interpreter commands.
+'''
+class QuitCmd(s_cli.CmdQuit):
+    '''
+    Quit the current command line interpreter.
+
+    Example:
+
+        !quit
+    '''
+    _cmd_name = '!quit'
+
+class HelpCmd(s_cli.CmdHelp):
+    '''
+    List interpreter extended commands and display help output.
+
+    Example:
+
+        !help foocmd
+
+    '''
+    _cmd_name = '!help'
+
+class StormCli(s_cli.Cli):
+
+    histfile = 'storm_history'
+
+    async def __anit__(self, item, outp=s_output.stdout, opts=None):
+
+        await s_cli.Cli.__anit__(self, item, outp=outp)
+
+        self.indented = False
+        self.cmdprompt = 'storm> '
+
+        self.stormopts = {'repr': True}
+        self.hidetags = False
+        self.hideprops = False
+
+        self.printf(welcome)
+
+    def initCmdClasses(self):
+        self.addCmdClass(QuitCmd)
+        self.addCmdClass(HelpCmd)
+
+    def printf(self, mesg, addnl=True, color=None):
+        if self.indented:
+            s_cli.Cli.printf(self, '')
+            self.indented = False
+        return s_cli.Cli.printf(self, mesg, addnl=addnl, color=color)
+
+    async def runCmdLine(self, line):
+
+        if line[0] == '!':
+            return await s_cli.Cli.runCmdLine(self, line)
+
+        async for mesg in self.item.storm(line, opts=self.stormopts):
```

Comment:
Instead of duplicating synaspe/cmds/cortex.py we could tease the common code out and re-use it?

Reply:
No, i dont want them to be tied together. there wont be much overlap by the time we're done accounting for callback vs elif and how local options are being stored anyway.